src/aof.c

```diff
@@ -1462,10 +1469,38 @@ int loadSingleAppendOnlyFile(char *filename) {
             ret = AOF_FAILED;
             goto cleanup;
         } else {
+            /* Restore the replication ID / offset from the RDB file. */
+            if (rsi.repl_id_is_set && rsi.repl_offset != -1 && rsi.repl_stream_db != -1) {
```

Comment:
This code is duplicated from `loadDataFromDisk()` where an RDB file is loaded. Let's avoid duplicated code by moving this common code to a function?